### PR TITLE
reactive: fix user edit

### LINKED ISSUE #14482: Editing user in Reactive application fails
##### **Overview of the issue**

If trying to edit and save user in Reactive application then error occurs and user is not saved.
Tried to change first name from **User** to **User 2**.
Stack trace:
```
{
  "type" : "https://www.jhipster.tech/problem/problem-with-message",
  "title" : "Internal Server Error",
  "status" : 500,
  "detail" : "executeMany; SQL [INSERT INTO jhi_user_authority VALUES(:userId, :authority)]; Unique index or primary key violation: \"PUBLIC.PRIMARY_KEY_E ON PUBLIC.JHI_USER_AUTHORITY(USER_ID, AUTHORITY_NAME) VALUES 3\"; SQL statement:\nINSERT INTO jhi_user_authority VALUES($1, $2) [23505-200]; nested exception is io.r2dbc.spi.R2dbcDataIntegrityViolationException: [23505] [23505] Unique index or primary key violation: \"PUBLIC.PRIMARY_KEY_E ON PUBLIC.JHI_USER_AUTHORITY(USER_ID, AUTHORITY_NAME) VALUES 3\"; SQL statement:\nINSERT INTO jhi_user_authority VALUES($1, $2) [23505-200]",
  "path" : "/api/admin/users",
  "message" : "error.http.500"
}
```

##### **Motivation for or Use Case**

It should be possible to edit users also in Reactive application.

##### **Reproduce the error**

* Generate application
* Run application in `dev` profile: `./gradlew`
* Go to Administration -> User Management
* Click **Edit** button in user **User** line
* Change first name from **User** to **User 2**
* Click **Save**

##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT C:\projects\teest\jwt_vs_oidc\jwt
`-- generator-jhipster@7.0.0  -> C:\projects\kaido-github\generator-jhipster

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "gateway",
    "baseName": "jhipster",
    "jhipsterVersion": "7.0.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1616862987472,
    "serviceDiscoveryType": false,
    "reactive": true,
    "authenticationType": "jwt",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "buildTool": "gradle",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "et",
    "packageFolder": "com/mycompany/myapp",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["et", "en", "ru"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

git version 2.31.0.windows.1

node: v14.16.0

npm: 6.14.12

Docker version 20.10.5, build 55c4c88

docker-compose version 1.28.5, build c4eb3a1f

##### **Browsers and Operating System**

Windows 10

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by chrisdns**
fix #14482 
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs [hunk 1]
```diff
@@ -269,6 +269,10 @@ public interface UserRepository extends R2dbcRepository<User, <% if (authenticat
 
     @Query("DELETE FROM <%= jhiTablePrefix %>_user_authority")
     Mono<Void> deleteAllUserAuthorities();
+
+    @Query("DELETE FROM jhi_user_authority WHERE user_id = :userId")
```
Comment (DanielFran) [inline generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs:273]:
```suggestion
    @Query("DELETE FROM <%= jhiTablePrefix %>_user_authority WHERE user_id = :userId")
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs (hunk 1 above)
Comment (chrisdns) [reply generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs]: thank you :+1:

**Comment by pascalgrimaud** [comment]
@chrisdns : you need to add the link of the open collective here, to confirm it is you, who submitted the invoice

**Comment by chrisdns** [comment]
https://opencollective.com/generator-jhipster/expenses/37188

**Pull request merged**